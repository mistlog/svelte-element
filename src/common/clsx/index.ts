/**
 * from https://github.com/lukeed/clsx
 */
function toVal(mix) {
	var k, y, str = '';

	if (typeof mix === 'string' || typeof mix === 'number') {
		str += mix;
	} else if (typeof mix === 'object') {
		if (Array.isArray(mix)) {
			for (k = 0; k < mix.length; k++) {
				if (mix[k]) {
					if (y = toVal(mix[k])) {
						str && (str += ' ');
						str += y;
					}
				}
			}
		} else {
			for (k in mix) {
				if (mix[k]) {
					str && (str += ' ');
					str += k;
				}
			}
		}
	}

	return str;
}

export type ClassValue = ClassArray | ClassDictionary | string | number | null | boolean | undefined;

export interface ClassDictionary {
	[id: string]: any;
}

export interface ClassArray extends Array<ClassValue> { }

export function Î£(...args: ClassArray) {
	var i = 0, tmp, x, str = '';
	while (i < arguments.length) {
		if (tmp = arguments[i++]) {
			if (x = toVal(tmp)) {
				str && (str += ' ');
				str += x
			}
		}
	}
	return str;
}
