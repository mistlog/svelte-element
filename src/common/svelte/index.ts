import { setContext as rawSetContext, getContext as rawGetContext } from "svelte";

export function setContext<T>(key: keyof T, value: T[keyof T]) {
    return rawSetContext(key, value);
}

export function getContext<T>(key: keyof T): T[keyof T] {
    return rawGetContext(key);
}

export type SvelteInternalProps = { $$slots?: any };
