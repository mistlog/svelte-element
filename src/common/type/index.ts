import { Readable, Writable } from "svelte/store";

export type InputOnChangeHandler = (event: Svelte.ChangeEvent<HTMLInputElement>) => void;
export type ButtonOnClickHandler = (event: Svelte.MouseEvent<HTMLButtonElement, MouseEvent>) => void;

type TypeWithGeneric<T> = Readable<T> | Writable<T>;
export type AutoSubscribe<Type> = Type extends TypeWithGeneric<infer X> ? X : never;

export type IconName = "edit" | "share" | "delete" | "search" | "arrow-left";