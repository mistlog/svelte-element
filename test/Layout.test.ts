import { render } from "@testing-library/svelte";
import { normalize } from "./common/utility";

import DefaultRow from "./assets/DefaultRow.svelte";
import RowGutter from "./assets/RowGutter.svelte";
import RowClassName from "./assets/RowClassName.svelte";
import DefaultCol from "./assets/DefaultCol.svelte";
import ColSpan from "./assets/ColSpan.svelte";
import ColOffset from "./assets/ColOffset.svelte";

describe("Row", () => {
    test("text only", () => {
        const { container } = render(DefaultRow);
        const row = container.querySelector(".el-row");
        expect(normalize(row.outerHTML)).toMatchSnapshot();
    });

    test("gutter", () => {
        const { container } = render(RowGutter);
        const col = container.querySelector(".el-col");
        expect(window.getComputedStyle(col).padding).toEqual("0px 5px");
    })

    test("className", () => {
        const { container } = render(RowClassName);
        const row = container.querySelector(".el-row");
        expect(window.getComputedStyle(row).display).toEqual("flex");

    })
})

describe("Col", () => {
    test("text only", () => {
        const { container } = render(DefaultCol);
        const col = container.querySelector(".el-col");
        expect(normalize(col.outerHTML)).toMatchSnapshot();
    });

    test("span", () => {
        const { container } = render(ColSpan);
        const col = container.querySelector(".el-col");
        expect(normalize(col.outerHTML)).toMatchSnapshot();
    })

    test("offset", () => {
        const { container } = render(ColOffset);
        const col = container.querySelector(".el-col");
        expect(normalize(col.outerHTML)).toMatchSnapshot();
    })
})