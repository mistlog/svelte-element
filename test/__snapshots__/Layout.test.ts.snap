// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col offset 1`] = `
"<div class=\\"el-col el-col-24 el-col-offset-6\\"></div>
"
`;

exports[`Col span 1`] = `
"<div class=\\"el-col el-col-12\\"></div>
"
`;

exports[`Col text only 1`] = `
"<div class=\\"el-col el-col-24\\">Default Col</div>
"
`;

exports[`Row text only 1`] = `
"<div class=\\"el-row\\">Default Row</div>
"
`;
